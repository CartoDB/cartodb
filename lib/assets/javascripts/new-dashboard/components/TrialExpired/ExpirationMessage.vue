<template>
  <section>
    <div class="container warning-container">
      <div class="warning-icon">
        <img src="../../assets/icons/common/warning-icon.svg" />
      </div>
      <h2 class="warning-title title is-title is-txtDarkBlue is-bold">
        {{ $t('TrialExpired.ExpirationMessage.title') }}
      </h2>
      <p class="text is-body is-txtDarkBlue">
         {{ this.trialExpiredMessage }}
      </p>
      <p class="warning-description text is-body is-txtDarkBlue"
         v-html="$t('TrialExpired.ExpirationMessage.description.phrase2', { expirationDays: this.expirationDays } )">
      </p>
      <div class="buttons-container">
        <a :href="addPaymentsURL" class="button upgrade-button is-caption">
          {{ $t('TrialExpired.ExpirationMessage.actions.addPayments') }}
        </a>
        <a @click="deleteAccount" class="button button--ghost is-txtDarkBlue is-caption">
          {{ $t('TrialExpired.ExpirationMessage.actions.deleteAccount') }}
        </a>
      </div>
    </div>
  </section>
</template>

<script>
import Dialog from 'new-dashboard/components/Backbone/Dialog';
import DeleteAccount from 'new-dashboard/components/Backbone/Dialogs/DeleteAccount.vue';
import format from 'date-fns/format';

export default {
  name: 'ExpirationMessage',
  props: {
    addPaymentsURL: String,
    expirationDays: {
      default: null
    }
  },
  computed: {
    trialExpiredMessage () {
      return (this.hasTrialExpirationDate
        ? this.$t('TrialExpired.ExpirationMessage.description.phrase1.withDate', { expirationDate: this.humanReadableExpirationDate })
        : this.$t('TrialExpired.ExpirationMessage.description.phrase1.noDate'));
    },
    hasTrialExpirationDate () {
      return this.$store.state.user.trial_ends_at;
    },
    humanReadableExpirationDate () {
      return format(this.$store.state.user.trial_ends_at, 'MMMM DD, YYYY');
    }
  },
  methods: {
    deleteAccount () {
      this.$modal.show({
        template: `
        <Dialog @close="$emit('close')">
          <DeleteAccount @close="$emit('close')"/>
        </Dialog>`,
        components: { Dialog, DeleteAccount }
      }, {}, { width: '100%', height: '100%' });
    }
  }
};
</script>

<style scoped lang="scss">
@import 'new-dashboard/styles/variables';

.upgrade-button {
  margin-right: 64px;
  background-color: $button-lockout__bg-color;
}

.link {
  text-decoration: underline;
}

.warning-container {
  text-align: center;
}

.warning-icon {
  margin-bottom: 48px;
}

.warning-title {
  margin-bottom: 16px;
}

.warning-description {
  margin-bottom: 16px;
}

.buttons-container {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 36px;
}
</style>
