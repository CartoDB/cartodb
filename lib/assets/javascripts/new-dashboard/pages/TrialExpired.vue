<template>
  <section class="page">
    <ExpirationMessage :expirationDays="expirationDays" :addPaymentsURL="addPaymentsURL"></ExpirationMessage>
  </section>
</template>

<script>
import ExpirationMessage from 'new-dashboard/components/TrialExpired/ExpirationMessage';

export default {
  name: 'TrialExpired',
  components: {
    ExpirationMessage
  },
  data () {
    return {
      expirationDays: window.expiration_days,
      addPaymentsURL: window.payments_url
    };
  }
};
</script>
