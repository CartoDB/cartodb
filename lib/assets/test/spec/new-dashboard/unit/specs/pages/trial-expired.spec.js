import { shallowMount } from '@vue/test-utils';
import TrialExpired from 'new-dashboard/pages/TrialExpired';

const $t = key => key;

describe('TrialExpired.vue', () => {
  it('should render correct contents', () => {
    const data = function () {
      return {
        expirationDays: 30,
        addPaymentsURL: 'https://carto.com/account/user/plan'
      };
    };

    const page = shallowMount(TrialExpired, {
      data,
      mocks: {
        $t
      }
    });

    expect(page).toMatchSnapshot();
  });
});
