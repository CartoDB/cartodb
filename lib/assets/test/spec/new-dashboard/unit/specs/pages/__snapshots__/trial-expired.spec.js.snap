// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrialExpired.vue should render correct contents 1`] = `
<section class="page">
  <expirationmessage-stub expirationdays="30" addpaymentsurl="https://carto.com/account/user/plan"></expirationmessage-stub>
</section>
`;
